pkg/controller/replicaset/replica_set.go

```diff
@@ -193,6 +194,44 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// getReplicaSetsWithSameController returns a list of ReplicaSets with the same
```

Comment:
this function needs a table driven test

Reply:
I didn't add it before because I figured the tests in `TestRelatedPodsLookup` for `getIndirectlyRelatedPods` (which uses `getReplicaSetsWithSameController`) would suffice, but I have now added `TestGetReplicaSetsWithSameController`: https://github.com/kubernetes/kubernetes/pull/80004/commits/de3f4955c09a7159974ccb8479e8edbd87c009e9#diff-16e152a1acd1373ada706d8cb4c099f6R351-R411